frontend/z3_axioms.py

```diff
@@ -388,6 +384,28 @@ def instance_axioms(called, args, result, types):
     return axioms
 
 
+def func_call(called, args, result, types):
+    if len(args) == 0:
+        return called == types.funcs[0](result)
+
+    subtype_axioms = []
+    z3_args = []
+    for i in range(len(args)):
+        z3_arg = getattr(types.type_sort, "func_{}_arg_{}".format(len(args), i + 1))(called)
+        z3_args.append(z3_arg)
+        arg = args[i]
+        subtype_axioms.append(And(
+            Implies(And(types.subtype(arg, types.num), types.subtype(z3_arg, types.num)),
```

Comment:
I don't understand this. Why do we have to use stronger_num in general constraints for function call parameters? That seems fishy. Even if it's not, there should be a comment in the code explaining what's going on.